calculator/calculator.py

```diff
@@ -1,15 +1,123 @@
 import re
+import math
+from typing import Union
+from statistics import mean
 
 
-class Calculator:
+_NUMBER_REGEX = r'\d+(?:\.\d+)?'
+_NUMBER_NEGATIVE_OPT_REGEX = f'-?{_NUMBER_REGEX}'
+_LEFT_PARENTHESIS_REGEX = r'(?P<LEFT_PARENTHESIS>\()'
+_RIGHT_PARENTHESIS_REGEX = r'(?P<RIGHT_PARENTHESIS>\))'
+_FACTORIAL_REGEX = r'(?P<FACTORIAL>!)'
+_NEGATE_REGEX = r'(?P<NEGATE>~)'
+_AVERAGE_REGEX = r'(?P<AVERAGE>@)'
+_MIN_REGEX = r'(?P<MIN>\&)'
+_MAX_REGEX = r'(?P<MAX>\$)'
+_POWER_REGEX = r'(?P<POWER>\^)'
+_MULTIPLICATION_REGEX = r'(?P<MULTIPLICATION>\*)'
+_DIVISION_REGEX = r'(?P<DIVISION>/)'
+_ADDITION_REGEX = r'(?P<ADDITION>\+)'
+_SUBTRACTION_REGEX = r'(?P<SUBTRACTION>-)'
```

Comment:
Why do you use named groups here?
You define handler to each regex, it's redundent information.